docs/release-guide.md

```diff
@@ -75,6 +12,51 @@ This guide requires:
  * GPG setup and personal key for signing
  * `cargo` (suggested: latest stable toolchain from [rustup][rustup])
  * `cargo-release` (suggested: `cargo install -f cargo-release`)
+ * A verified account on crates.io
+ * Write access to this GitHub project
+
+## Release checklist
+
+These steps show how to release version `x.y.z` on the `origin` remote (this can be determined via `git remote -av`).
+Push access to the upstream repository is required in order to publish the new tag and the PR branch.
+
+1. make sure the project is clean and prepare the environment:
+  - [ ] `cargo test`
+  - [ ] `cargo clean`
+  - [ ] `git clean -fd`
+  - [ ] `RELEASE_VER=x.y.z`
+  - [ ] `UPSTREAM_REMOTE=origin`
+
+:warning:: `UPSTREAM_REMOTE` should reference the locally configured remote that points to the upstream git repository.
+
+2. create release commits on a dedicated branch and tag it:
+  - [ ] `git checkout -b release-${RELEASE_VER}`
+  - [ ] `cargo release` (and confirm the version when prompted)
```

Comment:
For a minor release I had to run `cargo release -- minor`, as the `RELEASE_VER` environment variable is not considered by `cargo release`. Should we add a note on this or do we trust this to be intuitive enough?